src/renderer/library/components/layout/BreadCrumb.tsx

```diff
@@ -5,141 +5,192 @@
-                                                </Link>
-                                                <SVG ariaHidden svg={ChevronRight} />
-                                            </React.Fragment>
-                                            :
-                                        <strong key={index} title={item.name}>
-                                            {item.name}
-                                        </strong>,
-                                )}
-                            </>
-                        : <></> : <></>
-                } 
-            </div>
-        );
+const BreadCrumb = () => {
+
+    const spanLeft = React.useRef<HTMLSpanElement>(null);
+    const container = React.useRef<HTMLDivElement>(null);
+
+    const containerSize = useSize(container);
+    const containerWidth = Math.floor(containerSize?.width || -1);
+
+    const breadCrumbData = useSelector((state: ILibraryRootState) => state.opds.browser.breadcrumb);
+
+    const firstOne = {...(breadCrumbData.at(0) || {name: "", path: ""})};
+    // const lastOneFromBreadcrumbData = () => ({...(breadCrumbData.at(-1) || {name: "", path: ""})});
+    const lastOneFromBreadcrumbData = {...(breadCrumbData.at(-1) || {name: "", path: ""})};
+
+    const [lastOne, setLastOne] = React.useState(lastOneFromBreadcrumbData);
+
+    // React.useEffect(() => {
+    //     if (lastOne.name !== lastOneFromBreadcrumbData.name || lastOne.path !== lastOneFromBreadcrumbData.path) {
+    //         setLastOne({
+    //             name: lastOneFromBreadcrumbData.name,
+    //             path: lastOneFromBreadcrumbData.path,
+    //         });
+    //     }
+    // },
+    // [lastOne.name, lastOne.path, lastOneFromBreadcrumbData.name, lastOneFromBreadcrumbData.path]);
+
+    const between = [...(breadCrumbData.slice(1, -1) || [])];
+
+    // skip trim logic before initial useLayoutEffect or first resize observer
+    if (containerWidth > 0) {
+        do {
+            const sizeFootprint = breadcrumbLayoutSize(firstOne, between, lastOne);
+            if (sizeFootprint > containerWidth) {
+                between.shift();
+            } else {
+                break;
+            }
+        } while (between.length);
     }
-}
 
-const mapStateToProps = (state: ILibraryRootState) => ({
-    location: state.router.location,
-});
+    // I think it's not needed to run it inside an 'effect' because there are already a check of dependencies mutation
+    // React.useEffect(() => {
+    const spaceLeftWidth = Math.floor(spanLeft.current?.clientWidth || -1);
+    // const spaceLeftWidth = spanLeft.current?.clientWidth || -1;
+
+    // console.log(spaceLeftWidth, spanLeft.current?.clientWidth, containerWidth, between.length, firstOne.name.length, lastOne.name, lastOneFromBreadcrumbData.name);
+
+    if (between.length) {
+        if (lastOne.name !== lastOneFromBreadcrumbData.name || lastOne.path !== lastOneFromBreadcrumbData.path) {
```

Comment:
If I follow your logic, we can imitates the useEffect dependencies check by using the `object.is` property instead of  triple equal comparaison. What is the rational to not use the react.useEffect to do this ? Why react do not provide a mechanism to automatically do this instead of manually check, do you have an idea ? I'm thinking on it

Reply:
this check is because useState doesn't shallow-compare the object keys, just referential identity comparison.